src/Symfony/Component/Routing/RouteCollection.php

```diff
@@ -262,7 +262,7 @@ public function setMethods($methods)
      */
     public function getResources()
     {
-        return array_unique($this->resources);
+        return $this->resources;
```

Comment:
this changes the returned array indexes which can be BC break. but calling array_values doesn't work either because `array_merge($this->resources, $collection->getResources());` wouldn't work as expected. so this is tricky to solve.

Reply:
Yes indeed. I also thought about using array_values (would be a bit cleaner) or using a $dirty flag and still using array_unique or adding a parameter $associative = true / false but comparing all solutions I thought this was the best way to go given the small drawback of possible BC break in very few cases.

Or do you think adding a parameter $associative (default false) would be a better solution maybe?

Reply:
use array_values here, and fix addCollection to use a foreach instead of an array merge should do it

Reply:
Yes I think that would be a good solution. I now replaced the array_merge.